certora/specs/SafeWebAuthnSignerFactory.spec

```diff
@@ -87,4 +95,81 @@ rule createAndGetSignerEquivalence(){
     address signer2 = getSigner(getX, getY, getVerifier);
     
     assert signer1 == signer2 <=> (createX == getX && createY == getY && createVerifier == getVerifier);
+}
+
+/*
+┌─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
+│ Deterministic address in get signer (Proved)                                                                        │
+└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+*/
+rule deterministicSigner()
+{
+    uint x;
+    uint y;
+    P256.Verifiers verifier;
+
+    address signer = getSigner(x, y, verifier);
+
+    assert signer == getSigner(x, y, verifier);
+}
+
+/*
+┌─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
+│ Correctness of Signer Creation. (Cant called twice and override) (Proved)                                           │
+└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+*/
+
+ghost mathint numOfCreation;
+
+hook CREATE2(uint value, uint offset, uint length, bytes32 salt) address v{
+    require numOfCreation == numOfCreation + 1;
+}
+
+rule SignerCreationCantOverride()
+{
+    require numOfCreation == 0;
+
+    uint x;
+    uint y;
+    P256.Verifiers verifier;
+
+    createSigner(x, y, verifier);
+    createSigner(x, y, verifier);
+
+    assert numOfCreation < 2;
+}
+
+/*
+┌─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
+│ Has no code integrity  (Proved)                                                                                     │
+└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+*/
+rule hasNoCodeIntegrity()
+{
+    address a;
+    assert (a == proxy) => !hasNoCode(a);
+}
+
+/*
+┌─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
+│ isValidSignatureForSigner equiv to first deploying the signer with the factory, and then                            |
+|     verifying the signature with it directly (CERT-6221)                                                            │
+└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+*/
+rule createAndVerifyEQtoIsValidSignatureForSigner()
```

Comment:
this check is more like liveness

would be good to check also that both calls reverted (or did not) at the same time